### ISSUE TITLE
Remove jQuery.style/jQuery.css

### ISSUE THREAD

**Issue by mzgol** [jquery/jquery#2297]
`jQuery.style` & `jQuery.css` are undocumented, do we need to expose them or can we hide them as we plan to do with e.g. `buildFragment` & `domManip`?

**Comment by scottgonzalez** [comment]
We currently use `jQuery.css()` in jQuery UI. I'm sure we can switch to `.css()` for all of our uses though.

**Comment by gibson042** [comment]
Once upon a time, there was a ticket to make _more_ such methods: http://bugs.jquery.com/ticket/12851

I don't know if that's enough to save these, though.

**Comment by timmywil** [comment]
Yea, I think this is a bit different than `buildFragment` and `domManip`. I actually prefer using these methods sometimes. Forcing users to create a new jQuery collection to get/set CSS is unnecessary.

**Comment by dmethvin** [comment]
If the goal is to retroactively bless use of the current `jQuery.css()` and `jQuery.style()` as public APIs then we're going to have a pretty haphazard set of low-level APIs with inconsistent calling conventions. I guess it can't get any worse than `.each()` vs. `jQuery.each()` though. :smiling_imp: 

I'd rather have a `jQuery.css(elem, name, value)` setter and `jQuery.css(elem, name)` getter if we're going to have one, similar to `jQuery.data`. At least that would give us some consistency in the low level API, but it may break current users of the undocumented interface and we'd have to figure out what to do about the `extra` and `styles` args.

**Comment by mzgol** [comment]
> I'd rather have a jQuery.css(elem, name, value) setter and jQuery.css(elem, name) getter if we're going to have one, similar to jQuery.data. At least that would give us some consistency in the low level API, but it may break current users of the undocumented interface and we'd have to figure out what to do about the extra and styles args.

Definitely. I had a moment today when I was staring at `jQuery.style` & `jQuery.css` and not immediately seeing what's the responsibility division between them. This looks extremely confusing, especially when taking into account the fact that `jQuery.fn.css` is both a getter & a setter.

Having private positional params in public APIs also seems bad.

**Comment by timmywil** [comment]
Agreed. In this case, I'd rather see us move to what Dave suggested than drop them completely.

**Comment by mzgol** [comment]
> Agreed. In this case, I'd rather see us move to what Dave suggested than drop them completely.

That works for me as well.

**Comment by mzgol** [comment]
This relates to APIs currently undocumented so I assume deferring it to 3.1.0 would be ok?

**Comment by mzgol** [comment]
OK, deferred to 3.1.0. Please revert if you disagree.

**Comment by timmywil** [comment]
Moving to 4.0 for removal of jQuery.style, but to be extra safe.
